astropy/io/fits/tests/test_table.py

```diff
@@ -3270,6 +3270,72 @@ def test_empty_vla_raw_data(self):
                 [np.array([], dtype=np.int32), np.array([], dtype=np.int32)],
             )
 
+    def test_multidim_VLA_tables(self):
+        """
+        Check if multidimensional VLF are correctly write and read.
+        See https://github.com/astropy/astropy/issues/12860
+        and https://github.com/astropy/astropy/issues/7810
+        """
+        a = np.ones((5, 1))
+        b = np.ones((7, 1))
```

Comment:
Can you test with values that are not all 1, to be sure that they are correctly reshaped, saved etc. ?

Reply:
Done.